src/bundle/DependencyInjection/Configuration.php

```diff
@@ -93,74 +93,91 @@ public function getConfigTreeBuilder()
-                    ->children()
-                        ->arrayNode('identifiers')
-                            ->children()
-                                ->arrayNode('intro')
-                                    ->info('Content type intro field')
-                                    ->example('blog_post: blog_post_intro_field')
-                                    ->scalarPrototype()->end()
-                                ->end()
-                                ->arrayNode('image')
-                                    ->info('Content type image field')
-                                    ->example('article: article_image_field')
-                                    ->scalarPrototype()->end()
-                                ->end()
-                                ->arrayNode('author')
-                                    ->info('Content type author field')
-                                    ->example('article: article_author_field')
-                                    ->scalarPrototype()->end()
-                                ->end()
+            ->end()
+            ->arrayNode('field')
+                ->children()
+                    ->arrayNode('identifiers')
+                        ->children()
+                            ->arrayNode('intro')
+                                ->info('Content type intro field')
+                                ->example('blog_post: blog_post_intro_field')
+                                ->scalarPrototype()->end()
+                            ->end()
+                            ->arrayNode('image')
+                                ->info('Content type image field')
+                                ->example('article: article_image_field')
+                                ->scalarPrototype()->end()
+                            ->end()
+                            ->arrayNode('author')
+                                ->info('Content type author field')
+                                ->example('article: article_author_field')
+                                ->scalarPrototype()->end()
                             ->end()
                         ->end()
-                        ->arrayNode('relations')
-                        ->end()
+                    ->end()
+                    ->arrayNode('relations')
+                        ->beforeNormalization()
+                            ->ifTrue(
+                                function (array $relations) {
+                                    $pattern = '/\D+\.\D+/';
+                                    foreach ($relations as $relationFrom => $relationTo) {
+                                        return
+                                            !(
+                                                preg_match($pattern, $relationFrom)
+                                                && preg_match($pattern, $relationTo)
+                                            );
+                                    }
+                                }
```

Comment:
This does not check all relations. Instead, it will only check the first one.

What you probably wanted is to return early if this condition is false, and return true after foreach otherwise.

```suggestion
                                function (array $relations): bool {
                                    $pattern = '/\D+\.\D+/';
                                    foreach ($relations as $relationFrom => $relationTo) {
                                        if (!preg_match($pattern, $relationFrom) || !$preg_match($pattern, $relationTo)) {
                                            return true;
                                        }
                                    }
                                    
                                    return false;
                                }
```

I've also replaced the condition to be more readable. It's easier to wrap your head around simple negated conditions than to quickly read through the meaning of negated combined condition.

Reply:
An argument can also be made that `$relationFrom` and `$relationTo` could potentially be other types than `string`, so we also probably want to cast those before passing to `preg_match`.